nn_unet_scripts/convert_from_bids_to_nnunetv2_format.py

```diff
@@ -0,0 +1,329 @@
+    file_path : str
+        File path where the JSON file will be saved.
+    """
+    with open(file_path, "w") as f:
+        json.dump(data, f, indent=2)
+
+
+def process_images(
+    datapath: Path,
+    out_folder: str,
+    particpant_to_sample_dict: Dict[str, List[str]],
+    bids_to_nnunet_dict: Dict[str, int],
+    dataset_name: str,
+    is_test: bool = False,
+):
+    """
+    Processes all image files in each subject's directory.
+
+    Parameters
+    ----------
+    datapath : Path
+        Path to the data directory.
+    out_folder : str
+        Output directory to save processed images.
+    particpant_to_sample_dict : Dict[str, List[str]]
+        Dictionary mapping participant IDs to sample IDs.
+    bids_to_nnunet_dict : Dict[str, int]
+        Dictionary mapping subject names to case IDs.
+    dataset_name : str
+        Name of the dataset.
+    is_test : bool, optional
+        Boolean flag indicating if the images are for testing, by default False.
+    """
+    folder_type = "imagesTs" if is_test else "imagesTr"
+    image_suffix = "_0000"
+
+    for subject in particpant_to_sample_dict.keys():
+        for image in particpant_to_sample_dict[subject]:
+            case_id = bids_to_nnunet_dict[str((subject, image))]
+            image_path = os.path.join(
+                datapath, subject, "micr", f"{subject}_{image}_SEM.png"
+            )
+            img = cv2.imread(str(image_path), cv2.IMREAD_GRAYSCALE)
+            fname = f"{dataset_name}_{case_id:03d}{image_suffix}.png"
+            cv2.imwrite(os.path.join(out_folder, folder_type, fname), img)
+
+
+def process_labels(
+    datapath: Path,
+    out_folder: str,
+    particpant_to_sample_dict: Dict[str, List[str]],
+    bids_to_nnunet_dict: Dict[str, int],
+    dataset_name: str,
+    label_type: Literal["axonmyelin", "myelin", "axon"] = "axonmyelin",
+):
+    """
+    Processes label images from a list of subjects, matching each image with the label having the largest 'N' number.
+
+    Parameters
+    ----------
+    datapath : Path
+        Path to the data directory.
+    out_folder : str
+        Output directory to save processed label images.
+    particpant_to_sample_dict : Dict[str, List[str]]
+        Dictionary mapping participant IDs to sample IDs.
+    bids_to_nnunet_dict : Dict[str, int]
+        Dictionary mapping subject names to case IDs.
+    dataset_name : str
+        Name of the dataset.
+    """
+    label_type_to_divisor = {"axonmyelin": 127, "myelin": 255, "axon": 255}
+    for subject in particpant_to_sample_dict.keys():
+        for image in particpant_to_sample_dict[subject]:
+            case_id = bids_to_nnunet_dict[str((subject, image))]
+            label_path = os.path.join(
+                datapath,
+                "derivatives",
+                "labels",
+                subject,
+                "micr",
+                f"{subject}_{image}_SEM_seg-{label_type}-manual.png",
+            )
+            label = np.round(
```

Comment:
@hermancollin This is how I changed it. now we divide and round by 127 if we have two labels (axonmyelin) and otherwise we divide by 255.

Reply:
Ok this works!